src/axom/spin/internal/linear_bvh/BVH_impl.hpp

```diff
@@ -522,6 +620,99 @@ void BVH< NDIMS, ExecSpace, FloatType >::findRays( IndexType* offsets,
   axom::setDefaultAllocator( currentAllocatorID );
 }
 
+//------------------------------------------------------------------------------
+template< int NDIMS, typename ExecSpace, typename FloatType >
+void BVH< NDIMS, ExecSpace, FloatType >::findBoundingBoxes(
+  IndexType* offsets, IndexType* counts,
+  IndexType*& candidates, IndexType numBoxes,
+  const FloatType* xmin,
+  const FloatType* xmax,
+  const FloatType* ymin,
+  const FloatType* ymax,
+  const FloatType* zmin,
+  const FloatType* zmax ) const
+{
+  SLIC_ASSERT( offsets != nullptr );
+  SLIC_ASSERT( counts != nullptr );
+  SLIC_ASSERT( candidates == nullptr );
+  SLIC_ASSERT( xmin != nullptr );
+  SLIC_ASSERT( xmax != nullptr );
+  SLIC_ASSERT( ymin != nullptr );
+  SLIC_ASSERT( ymax != nullptr );
+
+  // STEP 0: set the default memory allocator to use for the execution space.
+  const int currentAllocatorID = axom::getDefaultAllocatorID();
+  const int allocatorID = axom::execution_space< ExecSpace >::allocatorID();
+  axom::setDefaultAllocator( allocatorID );
```

Comment:
Changing the global default allocator could be a problem for other libraries that are using a shared axom.
E.g. this could lead to concurrency issues which would be very difficult to track down.

Can you please create a ticket to modify the design of this class so that all allocators are local?

Tag: @gzagaris

Reply:
We are setting it back to the `currentAllocatorID` at the end of this function. I am addressing this in #134.